Jenkinsfile

```diff
@@ -163,19 +163,18 @@ node('docker && imgtec') {  // Only run on internal slaves as build takes a lot
                 [$class: 'FileBinding', credentialsId: 'opkg-build-private-key', variable: 'PRIVATE_KEY'],
                 [$class: 'FileBinding', credentialsId: 'opkg-build-public-key', variable: 'PUBLIC_KEY'],
             ]){
-                // Attempt to build quickly and reliably
+                sh "cp ${env.PRIVATE_KEY} ${WORKSPACE}/key-build"
+                sh "cp ${env.PUBLIC_KEY} ${WORKSPACE}/key-build.pub"
+
                 try {
-                    sh "cp ${env.PRIVATE_KEY} ${WORKSPACE}/key-build"
-                    sh "cp ${env.PUBLIC_KEY} ${WORKSPACE}/key-build.pub"
-                    sh "make -j4 V=s ${params.ALL_PACKAGES ? 'IGNORE_ERRORS=m' : ''}"
-                } catch (hudson.AbortException err) {
-                    // TODO BUG JENKINS-28822
-                    if(err.getMessage().contains('script returned exit code 143')) {
-                        throw err
+                    if(params.ALL_PACKAGES) {
+                        sh "make -j1 V=s IGNORE_ERRORS=m"
```

Comment:
-j4 causes packages to randomly not be built, depending where the build fails. This makes any opkg tests non-deterministic. There are two ways to do it in general, one like this, the other one is to collect the list of packages that failed to build and rerun builds just for them. This is the easier way at the moment. ;)

@Ham22 Can you chip in here if you disagree or there's anything else to add?